src/lib/draw/AttributeMode.svelte

```diff
@@ -60,44 +49,43 @@
     }
   }
 
-  function onMouseMove(e: MapMouseEvent) {
-    if (mode == thisMode) {
-      let results = $map.queryRenderedFeatures(e.point, {
-        layers: [
-          "interventions-points",
-          "interventions-lines",
-          "interventions-polygons",
-        ],
-      });
-      mapHover.set((results[0]?.id as number) || null);
-    }
-  }
+  const onMouseMove = (e: MapMouseEvent) => {
+    let results = $map.queryRenderedFeatures(e.point, {
+      layers: [
+        "interventions-points",
+        "interventions-lines",
+        "interventions-polygons",
+      ],
+    });
+    mapHover.set((results[0]?.id as number) || null);
+  };
 
-  function onMouseOut() {
-    if (mode == thisMode) {
-      mapHover.set(null);
-    }
-  }
+  const onMouseOut = () => {
+    mapHover.set(null);
+  };
 
-  function onClick(e: MapMouseEvent) {
-    if (mode == thisMode) {
-      let results = $map.queryRenderedFeatures(e.point, {
-        layers: [
-          "interventions-points",
-          "interventions-lines",
-          "interventions-polygons",
-        ],
-      });
-      if (results.length > 0) {
-        formOpen.set(results[0].id as number);
-      } else {
-        formOpen.set(null);
-      }
+  const onClick = (e: MapMouseEvent) => {
+    let results = $map.queryRenderedFeatures(e.point, {
+      layers: [
+        "interventions-points",
+        "interventions-lines",
+        "interventions-polygons",
+      ],
+    });
+    if (results.length > 0) {
+      formOpen.set(results[0].id as number);
+    } else {
+      formOpen.set(null);
     }
-  }
+  };
+
+  eventHandler.mapHandlers.mousemove = onMouseMove;
```

Comment:
I haven't tried yet, and it's not very important, but I wonder if we can't just do this directly and not need the intermediate `const`. From my understanding of closures in JS, it should be equivalent, with less boilerplate

Reply:
I wasn't able to see your think about preference between `function foo (bar){ ` and `const foo = (bar) =>{` but I don't really see one as preferable to the other. I guess consistency is preferable. I will try switch and see if it works

Reply:
Arrow functions are fine, I just meant could we do `eventHandler.mapHandlers.mousemove = (e: MapMouseEvent) => { ... }` and skip the intermediate const